cluster-operator/src/main/java/io/strimzi/operator/cluster/operator/assembly/KafkaAssemblyOperator.java

```diff
@@ -2053,78 +2053,88 @@ String zkConnectionString(int connectToReplicas, Function<Integer, String> zkNod
             for (GenericKafkaListener listener : loadBalancerListeners) {
                 String bootstrapServiceName = ListenersUtils.backwardsCompatibleBootstrapServiceName(name, listener);
 
-                Future perListenerFut = serviceOperations.hasIngressAddress(reconciliation, namespace, bootstrapServiceName, 1_000, operationTimeoutMs)
-                        .compose(res -> serviceOperations.getAsync(namespace, bootstrapServiceName))
-                        .compose(svc -> {
-                            String bootstrapAddress;
-
-                            if (svc.getStatus().getLoadBalancer().getIngress().get(0).getHostname() != null)    {
-                                bootstrapAddress = svc.getStatus().getLoadBalancer().getIngress().get(0).getHostname();
-                            } else {
-                                bootstrapAddress = svc.getStatus().getLoadBalancer().getIngress().get(0).getIp();
-                            }
-
-                            LOGGER.debugCr(reconciliation, "Found address {} for Service {}", bootstrapAddress, bootstrapServiceName);
+                List<String> listenerAddressList = new ArrayList<>(kafkaCluster.getReplicas() + 1);
```

Comment:
@scholzj I guess I probably need your help to take a look at this section...

I tested a couple of times about my PR and here is what I found:
1. In the new PR, the operator didn't wait for the completion of generating these Load Balancer Addresses, and it continue to generate the broker certificate and even create the pods with an empty Load Balancer address. Therefore the Kafka Broker can't start because of the lack of the Load Balancer address.
2. My suspicious is my understanding of the vert.x future is not correct, do I need a join after get the bootstrap address in line 2080?

Thanks!

Reply:
Yeah. The way you have it written, the `perListenerFut` will complete either on line 2057 or 2077. The whole block between lines 2082-2137 => it will not wait for it because it will be already complete. If you want, I can try to rewrite it tomorrow and share the code with you to compare it. Or we can have some call to go through it via screen sharing tomorrow (or when you have time). Whatever you prefer.

Reply:
Thank you for your time and the help. I tested locally, it works! Updating the PR now and adding some tests on it.

Reply:
@scholzj do you have any suggestions for the testing? I'm thinking to add the test in this file: https://github.com/strimzi/strimzi-kafka-operator/blob/main/cluster-operator/src/test/java/io/strimzi/operator/cluster/operator/assembly/KafkaAssemblyOperatorTest.java. It doesn't have the load balancer listener today, I guess I can add this type and test it with `createBootstrapService` disabled. Or it's better to do it in other places?

Reply:
No, please don't add it there. That parametrized test is quite bad since any test you add there will run in many unrelated combinations. Maybe `KafkaAssemblyOperatorNonParametrizedTest` would be a good place?